testonly/hammer/hammer.go

```diff
@@ -336,8 +355,17 @@ func (s *hammerState) pickKey(which int) []byte {
 		panic(fmt.Sprintf("internal error: can't pick a key, map contents[%d] is empty!", which))
 	}
 
-	choice := rand.Intn(l)
-	for k := range s.contents[which].data {
+	choice := s.prng.intN(l)
+	// Need sorted keys for reproduceability.
+	data := s.contents[which].data // map[mapKey]string
+	keys := make([]mapKey, 0)
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.SliceStable(keys, func(i, j int) bool {
+		return strings.Compare(data[keys[i]], data[keys[j]]) == -1
```

Comment:
This looks like it's sorting the keys based on contents. Did you want to sort keys based on keys?